### PR TITLE
test(side-menu): Adding unit tests

### PR DESCRIPTION
**Pull request by alfredskeden**
**Describe pull-request**  
Adds unit tests for side-menu component

**Solving issue**
Fixes: [CDEP-3010](https://tegel.atlassian.net/browse/CDEP-3010)

[CDEP-3010]: https://tegel.atlassian.net/browse/CDEP-3010?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core/src/components/side-menu/test/collapsible/side-menu.e2e.ts-snapshots/tds-side-menu-collapsible-renders-collapsible-side-menu-correctly-1-linux.png
Comment (mJarsater) [inline packages/core/src/components/side-menu/test/collapsible/side-menu.e2e.ts-snapshots/tds-side-menu-collapsible-renders-collapsible-side-menu-correctly-1-linux.png:1]: Might need to add some CSS styling in the index.html to allow for the side-menu to take full height

File: packages/core/src/components/side-menu/test/collapsible/side-menu.e2e.ts-snapshots/tds-side-menu-collapsible-click-collapse-button-to-close-the-menu-1-linux.png
Comment (mJarsater) [inline packages/core/src/components/side-menu/test/collapsible/side-menu.e2e.ts-snapshots/tds-side-menu-collapsible-click-collapse-button-to-close-the-menu-1-linux.png:1]: Might need to add some CSS styling in the index.html to allow for the side-menu to take full height

File: packages/core/src/components/side-menu/test/collapsible/side-menu.e2e.ts-snapshots/tds-side-menu-collapsible-renders-collapsible-side-menu-correctly-1-linux.png
Comment (theJohnnyMe) [reply packages/core/src/components/side-menu/test/collapsible/side-menu.e2e.ts-snapshots/tds-side-menu-collapsible-renders-collapsible-side-menu-correctly-1-linux.png:1]: I was thinking the same, but it is not only the body that needs styling. We would need to apply 100vh or similar tricks to the tds-side-menu too as that one depends on its surrounding elements. I checked our angular and react demo page and it is the same thing there.
Even thou screenshots might look broken, it is actually how the component looks in its purest form. Any additional styling to might cause false positives going forward as we test the part of the code that is not shipped with the component.

File: packages/core/src/components/side-menu/test/collapsible/side-menu.e2e.ts-snapshots/tds-side-menu-collapsible-click-collapse-button-to-close-the-menu-1-linux.png
Comment (theJohnnyMe) [reply packages/core/src/components/side-menu/test/collapsible/side-menu.e2e.ts-snapshots/tds-side-menu-collapsible-click-collapse-button-to-close-the-menu-1-linux.png:1]: Same as for the upper one. I liked the idea thou. ;)

**Review by theJohnnyMe — APPROVED** [review]

**Review by nathalielindqvist — APPROVED** [review]
LGTM

**Pull request merged**